src/System.Net.Security/src/System/Net/Security/NegotiateStream.cs

```diff
@@ -502,6 +502,11 @@ public override void Flush()
 #endif
         }
 
+        public override Task FlushAsync(CancellationToken cancellationToken)
+        {
+            return InnerStream.FlushAsync(cancellationToken);
```

Comment:
@stephentoub the Async test fails when I add the 
```
using (DebugThreadTracking.SetThreadKind(ThreadKinds.User | ThreadKinds.Async))
{
    await InnerStream.FlushAsync(cancellationToken);
}
```
And the task doesn't complete

Are they the wrong flags?